### PR TITLE
Context generator

### PR DESCRIPTION
**Pull request by djbe**
Use the "Generate Contexts" scheme to generate contexts instead of testing them. Also, had to update the pods --> use new PathKit methods to simplify some small things.

### CODE REVIEW

**Comment by djbe** [comment]
Need this for other PRs (so I don't have to do manually edit them all the time), so if we could get this merged soon(-ish)? 😆

**Comment by AliSoftware** [comment]
Having a dedicated _target_ which (if I'm not mistaken from reading that directly from GitHub) is a duplicate of the Test Target, has the risk of forgetting to add new test files when we have more tests and parsers in the future.

Wouldn't it be more maintainable to have the "Generate Contexts" _scheme_ which would build the Test target (the same as the existing scheme), so run the unit tests the same way… and use an **Environment Variable** or **Launch Arguments** to differentiate — instead of using `#if GENERATE_CONTEXTS`?

This way we would not need a separate target that would be the exact duplicate of the test target (except for that build setting) but instead still have one unique test target, and the main _scheme_ will run the test but the "Generate Contexts" scheme will run the exact same tests, but with an additional Launch Argument in the scheme.

_(We could even imagine instead to only keep one single scheme, not create a separate one, and have the Launch Argument pre-filled but unchecked, as we do for the Run action… but tbh I prefer having a separate scheme with an explicit name — as it would limit the risk of generating the contexts instead of testing them and make it more explicit for new contributors of what this action does, and that dedicated scheme can also only pilot the Test target but never be used for the App target)_

**Comment by djbe** [comment]
Using env. variables didn't work, so I settled for creating a scheme. Also created a rake task that goes along with it.

File: Tests/SwiftGenKitTests/TestsHelper.swift [hunk 1]
```diff
@@ -76,16 +76,16 @@ func XCTDiffContexts(_ result: [String: Any],
                      sub directory: Fixtures.Directory,
                      file: StaticString = #file,
                      line: UInt = #line) {
-  #if GENERATE_CONTEXTS
+  if ProcessInfo().environment["GENERATE_CONTEXTS"] == "YES" {
```
Comment (AliSoftware) [inline Tests/SwiftGenKitTests/TestsHelper.swift]: If I read the `xcscheme` properly, you gave it the value `Test` not the value `YES`, so you should check against that `Test` value not against `YES`?

Comment (djbe) [reply Tests/SwiftGenKitTests/TestsHelper.swift]: That was a mistake 😊

**Comment by djbe** [comment]
Follow up on our discussion from a while ago:
- Launch arguments didn't work for me. `xcodebuild` calls `xctest` with the following arguments, no matter what I tried from command line (nothing worked):
```["/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Xcode/Agents/xctest", "-NSTreatUnknownArgumentsAsOpen", "NO", "-ApplePersistenceIgnoreState", "YES"]```
- We could just do env. variables from rake, but then users wouldn't be able to generate contexts from Xcode.

Thoughts? I think we might as well merge this.

**Review by AliSoftware — COMMENTED** [review]
Only thing missing is some mention of this in a CONTRIBUTING.md or similar, and after that it's good to merge!

**Pull request merged**